### PR TITLE
Commerce 9494 2

### PR DESCRIPTION
**Pull request by MahmoudAzzam**
https://issues.liferay.com/browse/COMMERCE-9494

### CODE REVIEW

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/resource/v1_0/WishListResourceImpl.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+ * @author Mahmoud Azzam
+ */
+@Component(
+	properties = "OSGI-INF/liferay/rest/v1_0/wish-list.properties",
+	scope = ServiceScope.PROTOTYPE, service = WishListResource.class
+)
+public class WishListResourceImpl extends BaseWishListResourceImpl {
+
+	@Override
+	public Response deleteWishList(Long wishListId) throws Exception {
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/resource/v1_0/WishListResourceImpl.java:46]: delete should return void

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/resource/v1_0/WishListResourceImpl.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+	}
+
+	@Override
+	public WishList patchChannelWishList(
+			Long channelId, Long accountId, Long wishListId, WishList wishList)
+		throws Exception {
+
+		return _toWishList(
+			_commerceWishListService.updateCommerceWishList(
+				wishListId, wishList.getName(), wishList.getDefaultWishList()));
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/resource/v1_0/WishListResourceImpl.java:89]: use getterutil and the defualt value will be the already saved value for both name and default wishlist

Comment (andrea-ale-sbarra) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/resource/v1_0/WishListResourceImpl.java:89]: also you're not using channelid and accountid so remove parameters

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/resource/v1_0/WishListResourceImpl.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+	public WishList postChannelWishList(
+			Long channelId, Long accountId, WishList wishList)
+		throws Exception {
+
+		CommerceChannel commerceChannel =
+			_commerceChannelService.getCommerceChannel(channelId);
+
+		return _toWishList(
+			_commerceWishListService.addCommerceWishList(
+				wishList.getName(), wishList.getDefaultWishList(),
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/resource/v1_0/WishListResourceImpl.java:102]: getter util

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/resource/v1_0/WishListResourceImpl.java (hunk 2 above)
Comment (andrea-ale-sbarra) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/resource/v1_0/WishListResourceImpl.java:89]: so also the endpoint is wrong

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/resource/v1_0/WishListResourceImpl.java [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+					pagination.getStartPosition(), pagination.getEndPosition(),
+					null),
+				commerceWishList -> _toWishList(commerceWishList)),
+			pagination,
+			_commerceWishListService.getCommerceWishListsCount(
+				commerceChannel.getSiteGroupId(), contextUser.getUserId()));
+	}
+
+	@Override
+	public WishList getWishList(Long channelId, Long accountId, Long wishListId)
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/resource/v1_0/WishListResourceImpl.java:75]: channelid and accountid are not used

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/resource/v1_0/WishListItemResourceImpl.java [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+	service = {NestedFieldSupport.class, WishListItemResource.class}
+)
+public class WishListItemResourceImpl
+	extends BaseWishListItemResourceImpl implements NestedFieldSupport {
+
+	@Override
+	public Response deleteWishListItem(Long wishListItemId) throws Exception {
+		_commerceWishListItemService.deleteCommerceWishListItem(wishListItemId);
+
+		Response.ResponseBuilder responseBuilder = Response.noContent();
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/resource/v1_0/WishListItemResourceImpl.java:65]: return void

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]
Please try to understand if you can get channel and account from the wishlist and if you can get the wishlist from the wishlistitem

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
looks good to me now, thank you @MahmoudAzzam !

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/dto/v1_0/converter/WishListDTOConverter.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+		CommerceWishList commerceWishList =
+			_commerceWishListService.getCommerceWishList(
+				(Long)dtoConverterContext.getId());
+
+		return new WishList() {
+			{
+				defaultWishList = commerceWishList.isDefaultWishList();
+				id = commerceWishList.getCommerceWishListId();
+				name = commerceWishList.getName();
```
[the remark this answers is not in the document]
Comment (MahmoudAzzam) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/dto/v1_0/converter/WishListDTOConverter.java:53]: @andrea-ale-sbarra I do not think there is

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/rest-openapi.yaml [hunk 7]
```diff
@@ -1759,5 +1792,624 @@ paths:
...
+                            schema:
+                                $ref: "#/components/schemas/Error"
+                        application/xml:
+                            schema:
+                                $ref: "#/components/schemas/Error"
+                    description:
+                        Unexpected error
+            summary: Creates a WishList Item.
+            tags: ["WishListItem"]
+    /channels/{channelId}/wishlists/{wishListId}/wishlist-items/{wishListItemId}:
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/rest-openapi.yaml:2135]: after reviewing this with @luca-pellizzon I believe you don't need to pass neither the channel id nor the wishlistId, you can find those info from the wishlistitem, you can fetch the wishlist from the item directly and you can fetch the channel with channel service looking for sitegroup id

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]
also looks like the get endpoint of the single wish list is missing, can you add it if this is true?

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/rest-openapi.yaml (hunk 7 above)
Comment (MahmoudAzzam) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/rest-openapi.yaml:2135]: @andrea-ale-sbarra I removed both channelId and wishListId but I think there is a bug that stopping me from proceeding. 
The getCommerceChannelByGroupId method ends up calling commerceChannelLocalService.getCommerceChannel(
group.getClassPK()) which expects channelId but instead we are passing groupId

My code is as follows:
<<<<<<<<<<<<
public WishListItem getWishListItem(
			Long accountId, Long wishListItemId)
throws Exception {

CommerceWishListItem commerceWishListItem =
	_commerceWishListItemService.getCommerceWishListItem(
		wishListItemId);

CommerceChannel commerceChannel =
	_commerceChannelLocalService.getCommerceChannelByGroupId(
		commerceWishListItem.getGroupId());

CommerceContext commerceContext = _commerceContextFactory.create(
	commerceChannel.getCompanyId(), commerceChannel.getGroupId(),
	commerceChannel.getUserId(), 0,
	_getCommerceAccountId(accountId, commerceChannel));

return _toWishListItem(commerceWishListItem, commerceContext);
}
>>>>>>>>>>>>

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/rest-openapi.yaml [hunk 8]
```diff
@@ -1759,5 +1792,624 @@ paths:
...
+                            schema:
+                                $ref: "#/components/schemas/Error"
+                    description:
+                        Authentication information is missing or invalid
+            summary: Deletes a WishList by ID.
+            tags: ["WishList"]
+        get:
+            description:
+                Retrieves a wishlist by wishListId.
+            operationId: getWishList
```
Comment (MahmoudAzzam) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/rest-openapi.yaml:2292]: > also looks like the get endpoint of the single wish list is missing, can you add it if this is true?
@andrea-ale-sbarra you can find it here

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/rest-openapi.yaml (hunk 7 above)
Comment (MahmoudAzzam) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/rest-openapi.yaml:2135]: @andrea-ale-sbarra 
Please note that the code above is not in this PR

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/resource/v1_0/WishListItemResourceImpl.java [hunk 9]
```diff
@@ -0,0 +1,237 @@
...
+	@Override
+	public Page<WishListItem> getWishListItemsPage(
+			@NestedFieldId("id") Long wishListId, Long accountId,
+			Pagination pagination)
+		throws Exception {
+
+		CommerceWishList commerceWishList =
+			_commerceWishListService.getCommerceWishList(wishListId);
+
+		CommerceChannel commerceChannel =
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/resource/v1_0/WishListItemResourceImpl.java:101]: null check for channel

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/resource/v1_0/WishListItemResourceImpl.java [hunk 10]
```diff
@@ -0,0 +1,237 @@
...
+		String cpInstanceUuid = StringPool.BLANK;
+
+		if (cpInstance != null) {
+			cpInstanceUuid = cpInstance.getCPInstanceUuid();
+		}
+
+		CommerceWishList commerceWishList =
+			_commerceWishListService.getCommerceWishList(wishListId);
+
+		CommerceChannel commerceChannel =
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/resource/v1_0/WishListItemResourceImpl.java:139]: null check for channel

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/resource/v1_0/WishListResourceImpl.java [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+	public void deleteWishList(Long wishListId) throws Exception {
+		_commerceWishListService.deleteCommerceWishList(wishListId);
+	}
+
+	@Override
+	public Page<WishList> getChannelWishListsPage(
+			Long channelId, Long accountId, Pagination pagination)
+		throws Exception {
+
+		CommerceChannel commerceChannel =
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/resource/v1_0/WishListResourceImpl.java:54]: null check for channel

Comment (andrea-ale-sbarra) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/resource/v1_0/WishListResourceImpl.java:54]: or use getChannel method

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

**Review by andrea-ale-sbarra — APPROVED** [review]
Good, thank you @MahmoudAzzam

**Comment by andrea-ale-sbarra** [comment]
ci:test:commerce

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

**Comment by MahmoudAzzam** [comment]
@andrea-ale-sbarra please review the latest change

**Review by andrea-ale-sbarra — APPROVED** [review]
thank you @MahmoudAzzam

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by baharturk** [comment]
ci:test:commerce

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

**Pull request closed without merging**